backend/src/exercise/http-handler/api/exercise.ts

```diff
@@ -1,25 +1,93 @@
+import { plainToInstance } from 'class-transformer';
 import type { ExerciseIds, ExerciseTimeline } from 'digital-fuesim-manv-shared';
-import { ExerciseState } from 'digital-fuesim-manv-shared';
+import {
+    StateExport,
+    ExerciseState,
+    validateExerciseExport,
+    ReducerError,
+} from 'digital-fuesim-manv-shared';
+import { isEmpty } from 'lodash-es';
 import type { DatabaseService } from '../../../database/services/database-service';
 import { UserReadableIdGenerator } from '../../../utils/user-readable-id-generator';
 import { exerciseMap } from '../../exercise-map';
 import { ExerciseWrapper } from '../../exercise-wrapper';
 import type { HttpResponse } from '../utils';
 
-export function postExercise(
-    databaseService: DatabaseService
-): HttpResponse<ExerciseIds> {
+export async function postExercise(
+    databaseService: DatabaseService,
+    importObject: StateExport
+): Promise<HttpResponse<ExerciseIds>> {
     let newParticipantId: string | undefined;
     let newTrainerId: string | undefined;
     try {
         newParticipantId = UserReadableIdGenerator.generateId();
         newTrainerId = UserReadableIdGenerator.generateId(8);
-        const newExercise = ExerciseWrapper.create(
-            newParticipantId,
-            newTrainerId,
-            databaseService,
-            ExerciseState.create()
-        );
+        let newExercise: ExerciseWrapper;
```

Comment:
Instead of `let` and assignment in try catch, you could also put the potentially throwing code in an extra function that catches it and returns undefined/another value if it fails.

-> less nesting and const instead of let